Dockerfile.dev

```diff
@@ -0,0 +1,18 @@
+FROM golang:1.19-alpine
+
+WORKDIR /app/github.com/cryptag/leapchat/miniware
+
+WORKDIR /app/github.com/cryptag/leapchat
+
+RUN mkdir -p /build
+
+COPY miniware/* miniware/
+COPY *.go .
+COPY go.* .
+COPY LICENSE.md .
+
+RUN go build -o /leapchat
+
+EXPOSE 8080
+
+CMD ["/leapchat"]
```

Comment:
@elimisteve So you're right, `npm run dev` just does a build to the `/build` directory. The leapchat binary just serves content directly from there.

I removed the Docker frontend container / file. I don't think it makes sense right here, given the current workflow.

Instead, in the immediate-term, I think we can create a volume in this container that binds `build` to a directory next to the binary in the container? `npm start` would then run the `watch-build` loop and update the `build` dir in the go container.

Reply:
@jimmcgaw Removing that other container and mounting `build` into the Go container sounds perfect.

The only detail I'd add is: instead of putting the `leapchat` Go binary at `/leapchat`, let's put it next to the other *.go files just so that `leapchat` is running in the same environment everywhere: in Docker, on my laptop, and on the production server. I mention that here because this implies putting the `build` directory right next to `leapchat` and thus inside the `/app/github.com/cryptag/leapchat` directory.

Sound good?

Reply:
In that case I think we just need this:

```dockerfile
FROM golang:1.19-alpine

RUN mkdir -p /app/github.com/cryptag/leapchat/miniware

WORKDIR /app/github.com/cryptag/leapchat/

COPY miniware/* miniware/
COPY *.go .
COPY go.* .
COPY LICENSE.md .

RUN go build

EXPOSE 8080

CMD ["./leapchat"]
```

And then to mount `./build` as `/app/github.com/cryptag/leapchat/build` inside the container. (I believe `/app/github.com/cryptag/leapchat/build` doesn't need to and indeed probably shouldn't exist inside the container, since mounting a volume to that directory would overwrite it or maybe not work?)

Reply:
Actually, if this works we can make it more like you suggested above, with a simpler directory structure in the container:

```dockerfile
FROM golang:1.19-alpine

RUN mkdir -p /leapchat/miniware

WORKDIR /leapchat

COPY miniware/* miniware/
COPY *.go .
COPY go.* .
COPY LICENSE.md .

RUN go build

EXPOSE 8080

CMD ["./leapchat"]
```